packages/web/src/components/search/addons/ImageDropdown.js

```diff
@@ -0,0 +1,297 @@
+ *
+ * The file browser maintains an internal state of the file
+ * which is accesible by input.files, where input is the file input element.
+ * When the file input element is unmounted this internal state is lost.
+ * Hence we should always use the imageValue provided by the user for the preview image
+ * On delete, we should also call onChange provided by the user with an empty string.
+ *
+ * 2. Drag and drop
+ *
+ * 3. URL
+ * */
+// eslint-disable-next-line import/prefer-default-export
+export const ImageDropdown = ({ imageValue, onChange, onOutsideClick }) => {
+	const fileInputRef = useRef();
+	const imageRef = useRef();
+	const containerRef = useRef();
+	const [url, setURL] = useState('');
+	const [showError, setShowError] = useState(false);
+	const [isDropping, setIsDropping] = useState(false);
+	const urlValueTimer = useRef(null);
+
+	const handleDelete = () => {
+		const image = imageRef.current;
+		if (image) {
+			onChange('');
+			setURL('');
+			setShowError(false);
+		}
+	};
+
+	const readFile = (file) => {
+		const reader = new FileReader();
+
+		reader.addEventListener(
+			'load',
+			() => {
+				const img = document.createElement('img');
+				img.onload = () => {
+					const MAX_WIDTH = 512;
+					const MAX_HEIGHT = 512;
+
+					let width = img.width;
+					let height = img.height;
+
+					// Change the resizing logic
+					if (width > height) {
+						if (width > MAX_WIDTH) {
+							height *= (MAX_WIDTH / width);
+							width = MAX_WIDTH;
+						}
+					} else if (height > MAX_HEIGHT) {
+						width *= (MAX_HEIGHT / height);
+						height = MAX_HEIGHT;
+					}
+
+					// Dynamically create a canvas element
+					const canvas = document.createElement('canvas');
+
+					canvas.width = width;
+					canvas.height = height;
+
+					// const canvas = document.getElementById("canvas");
+					const ctx = canvas.getContext('2d');
+
+					// Actual resizing
+					ctx.drawImage(img, 0, 0, width, height);
+
+					// Show resized image in preview element
+					const dataurl = canvas.toDataURL(file.type);
+
+					// Below would change the preview if the user changes the imageValue respecting below
+					onChange(dataurl);
+					setShowError(false);
+				};
+				img.onerror = () => {
+					if (!img.width) {
+						onChange('');
+						setShowError(true);
+					}
+				};
+
+				if (!reader.result) {
+					setShowError(true);
+					return;
+				}
+				img.src = reader.result;
+			},
+			false,
+		);
+		reader.addEventListener('error', () => {
+			setShowError(true);
+		});
+
+		if (file) {
+			reader.readAsDataURL(file);
+		} else {
+			onChange('');
+		}
+	};
+
+	useEffect(() => {
+		const imagePreview = imageRef.current;
+		if (imagePreview && imageValue) {
+			const image = new Image();
+			image.onerror = () => {
+				if (!image.width) {
+					onChange('');
+					setShowError(true);
+				}
+			};
+			image.src = imageValue;
+		}
+	}, [imageValue]);
+
+	useEffect(() => {
+		window.addEventListener('click', (e) => {
```

Comment:
@SavvyShah Do you think we should remove the listenere?

Reply:
@SavvyShah Please address the comments

Reply:
Already addressed.